System.IO.Ports/SerialPort.cs

```diff
@@ -406,15 +406,14 @@ public int ReadTimeout
         /// <exception cref="IOException">The port is in an invalid state. -or- An attempt to set the state of the underlying
         /// port failed. For example, the parameters passed from this <see cref="SerialPort"/>
         /// object were invalid.</exception>
-        /// <exception cref="ArgumentOutOfRangeException">The <see cref="WriteTimeout"/> value is less than zero and not equal
-        /// to <see cref="Timeout"/>.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">The read time-out value is less than zero and not equal to <see cref="Timeout.Infinite"/>.</exception>
         public int WriteTimeout
         {
             get => _writeTimeout;
 
             set
             {
-                if ((value < 0) && (value != Timeout.Infinite))
+                if (value is < 0 && value != Timeout.Infinite)
```

Comment:
Same goes for here...

Reply:
That was there but our style cop rules are old school and don't accept that. That's why I reverted it to the version you're seeing.

Reply:
Given comments I have received (in the past)... I think it is best to ignore some `StyleCop` rules with most of the core libraries... and definitely not C#9... (although it does have its uses!)... but... (and with my internal work) SonarCloud is better (if you can get it to throw warnings via output to GH WF)!

Reply:
Infact... #103 is a good reason as to why they should be ignored...

Reply:
It's deeper than that. We can't alleviate that rule, because it's not a rule. This error comes up when there is an internal error in StyleCop. Reason being we're forced to use an older version of it, pre C# 9.0, therefore it finds this as an error.
Remedy would be to exclude the file from check, but that would defeat the purpose of it.

Look [here](https://stackoverflow.com/questions/20563385/how-to-suppress-stylecop-error-sa0102-csharp-csparser-a-syntax-error-has-bee) for more details.

At the end of the day, this is syntactic, so no harm on the actual outcome.